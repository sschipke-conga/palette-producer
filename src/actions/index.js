

export const setUser = user => ({
  type: 'SET_USER',
  user
});

export const setAllProjects = allProjects => ({
  type: 'SET_PROJECTS',
  allProjects
});

export const setCurrentPalette = palette => ({
  type: 'SET_CURRENT_PALETTE',
  palette
});

export const setCurrentPaletteInfo = info => ({
  type: 'SET_CURRENT_PALETTE_INFO',
  info
});

export const setCurrentProjectInfo = info => ({
  type: 'SET_CURRENT_PROJECT_INFO',
  info
});

export const setAllPalettes = allPalettes => ({
  type: 'SET_PALETTES',
  allPalettes
});

export const toggleMenu = () => ({
  type: 'TOGGLE_MENU'
})

export const removePalette = (id) => ({
  type: 'REMOVE_PALETTE',
  id: parseInt(id)
});