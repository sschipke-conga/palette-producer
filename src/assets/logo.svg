<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" width="316.9996" height="316.9996" viewBox="0 0 316.9996 316.9996"><defs><style>.cls-1{fill:#fff;}.cls-1,.cls-2,.cls-3,.cls-4,.cls-5{stroke:#000;stroke-miterlimit:10;stroke-width:10px;}.cls-2{fill:#fbb03b;}.cls-3{fill:#f15a24;}.cls-4{fill:#d4145a;}.cls-5{fill:#3c4892;}</style></defs><title>Artboard 5</title><path class="cls-1" d="M31.62037,143.03632c.33314-55.44444,42.19532-108.237,95.8371-120.86067C185.4136,8.53673,248.78037,35.47176,273.57154,85.89533c31.15582,63.36876,1.52388,129.99623-69.57942,114.45468-13.68126-2.99042-12.8967-5.45845-31.15819-10.9585-23.80811-7.1706-24.20607,3.48171-14.73851,56.5,5,28,7,39-8,49C111.92072,320.34136,31.22091,209.51685,31.62037,143.03632Z"/><circle class="cls-2" cx="75.01795" cy="138.12396" r="22.92254"/><circle class="cls-3" cx="116.7926" cy="76.84227" r="22.92254"/><circle class="cls-4" cx="195.10246" cy="76.8141" r="22.92254"/><circle class="cls-5" cx="230.17288" cy="147.88452" r="22.92254"/></svg>