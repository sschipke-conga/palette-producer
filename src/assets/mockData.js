export const mockPalettes = [
  {
    id: 30,
    project_id: 35,
    name: "My Second Palette",
    color1: "#e5349d",
    color2: "#bdaac5",
    color3: "#5a9def",
    color4: "#2705b2",
    color5: "#309a6e",
    created_at: "2020-01-21T19:09:20.992Z",
    updated_at: "2020-01-21T19:09:20.992Z",
  },
  {
    id: 33,
    project_id: 35,
    name: "New one now",
    color1: "#f36d6b",
    color2: "#72cc20",
    color3: "#f11a21",
    color4: "#8eb4e8",
    color5: "#b08980",
    created_at: "2020-02-15T00:06:46.617Z",
    updated_at: "2020-02-15T00:06:46.617Z",
  },
  {
    id: 40,
    project_id: 35,
    name: "This time it should work",
    color1: "#57101d",
    color2: "#324086",
    color3: "#5a789c",
    color4: "#959191",
    color5: "#18a647",
    created_at: "2020-02-19T03:52:55.467Z",
    updated_at: "2020-02-19T03:52:55.467Z",
  },
  {
    id: 45,
    project_id: 37,
    name: "My Second Palette",
    color1: "#e5349d",
    color2: "#bdaac5",
    color3: "#5a9def",
    color4: "#2705b2",
    color5: "#309a6e",
    created_at: "2020-01-21T19:09:20.992Z",
    updated_at: "2020-01-21T19:09:20.992Z"
  },
  {
    id: 43,
    project_id: 37,
    name: "This time it should work",
    color1: "#57101d",
    color2: "#324086",
    color3: "#5a789c",
    color4: "#959191",
    color5: "#18a647",
    created_at: "2020-02-19T03:52:55.467Z",
    updated_at: "2020-02-19T03:52:55.467Z"
  }
]

export const mockProjects = [
  {
    id: 30,
    name: "My First Project",
  },
  {
    id: 33,
    name: "Second Projects",
  },
  {
    id: 40,
    name: "Third Project"
  }
]

export const mockUser = {
  user_id: 34,
  username: 'Gary'
}

export const mockCurrentPalette = [
  {
    name: 'color1',
    hexCode: '#FFFFFF',
    isLocked: false
  },
  {
    name: 'color2',
    hexCode: '#FFFFFF',
    isLocked: false
  },
  {
    name: 'color3',
    hexCode: '#FFFFFF',
    isLocked: false
  },
  {
    name: 'color4',
    hexCode: '#FFFFFF',
    isLocked: false
  },
  {
    name: 'color5',
    hexCode: '#FFFFFF',
    isLocked: false
  }
]