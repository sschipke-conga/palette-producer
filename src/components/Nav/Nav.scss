nav {
  border-bottom: 2px solid grey;
  display: flex;
  justify-content: space-between;
  padding: 20px;
  a {
    border-left: 1px solid gray;
    color: black;
    padding: 10px 10px 10px 20px;
    text-decoration: none;
    &:hover {
      opacity: 0.5;
    }
  }
}