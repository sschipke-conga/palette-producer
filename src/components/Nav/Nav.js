import React from 'react';
import { NavLink } from 'react-router-dom';
import './Nav.scss'
import logo from '../../assets/2026954.svg';

const Nav = () =>
  <nav>
    <img className="logo" src={logo} alt="logo" />
    <NavLink exact to="/signup">Account</NavLink>
  </nav>

export default Nav;