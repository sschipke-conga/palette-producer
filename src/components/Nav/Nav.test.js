import React from "react";
import { Nav, mapStateToProps, mapDispatchToProps} from "./Nav";
import { shallow } from "enzyme";
import {toggleMenu} from '../../actions/index'
import {mockUser} from '../../assets/mockData'


describe("Nav", () => {
  const mockToggleMenu = jest.fn()
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Nav
      toggleMenu={mockToggleMenu}
      user={mockUser}
      />
    );
  });
  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should call toggleMenu when the  button is clicked', () => {
    wrapper.find('button').simulate('click');
    expect(mockToggleMenu).toHaveBeenCalled()
  })
  describe('mapStateToProps/mapDispatchToProps', () => {
    it('mapStateToProps the user and the state of the menu', () => {
      const mockUser = { username: 'Dave', id: 4 }
      const mockState = {
        user: mockUser,
        isMenuActive: false
      };
      const expected = {
        user: mockUser,
        isMenuActive: false
      };
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    });

    it('calls dispatch with setAllPallettes action when it is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = toggleMenu('TOGGLE_MENU', true);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.toggleMenu('SET_USER', true);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  })
});