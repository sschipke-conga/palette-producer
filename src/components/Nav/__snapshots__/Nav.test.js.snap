// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
<nav>
  <img
    alt="logo"
    className="logo"
    src="logo.svg"
  />
  <button
    onClick={[MockFunction]}
  >
    Menu
  </button>
  <div>
    <p
      className="display-username"
    >
      Welcome, 
      Gary
      !
    </p>
  </div>
</nav>
`;
